cpp/dolfin/function/Constant.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2019 Chris Richardson, Michal Habera
+//
+// This file is part of DOLFIN (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#pragma once
+
+#include <vector>
+
+namespace dolfin
+{
+
+namespace function
+{
+
+class Constant
+{
+public:
+  /// Initialise as a vector
+  ///
+  /// The vector is a row-major (C style) flattened value of the constant
+  explicit Constant(std::vector<PetscScalar> value) : value(value){};
```

Comment:
Should be include the rank/shape? Would be more expressive.

Would be nice to support the common case of rank 0 (`PETScScalar`), 1 (`std::vector` or `Eigen::Vector`) and 2 (`Eigen::Matrix`).